Sources/MotorControl/ViewModel/MotionSensorViewModels.swift

```diff
@@ -131,6 +82,51 @@ public final class MotionSensorStepViewModel : AbstractMotionControlState {
     }
 }
 
+/// View model for a tremor step
+public final class TremorStepViewModel : MotionSensorStepViewModel {
+}
+
+/// View model for a tapping step
+public final class TappingStepViewModel : MotionSensorStepViewModel {
+    public var samples: [TappingSample] = []
+    public var lastSample: [TappingButtonIdentifier : TappingSample] = [:]
+    public var previousButton: TappingButtonIdentifier? = nil
+    @Published public var tapCount: Int = 0
+    @Published public var isPaused : Bool = false {
+        didSet {
+            guard recorder.status >= .starting else { return }
+            if isPaused {
+                recorder.pause()
+            }
+            else {
+                recorder.resume()
+            }
+        }
+    }
+    public var initialTap: Bool { recorder.status > .idle }
+
+    public func tappedScreen(uptime: TimeInterval,
+                             timestamp: TimeInterval,
+                             currentButton: TappingButtonIdentifier,
+                             location: CGPoint,
+                             duration: TimeInterval) {
+        let sample: TappingSample = .init(uptime: uptime - duration,
+                                          timestamp: timestamp - duration,
+                                          stepPath: recorder.currentStepPath,
+                                          buttonIdentifier: currentButton,
+                                          location: location,
+                                          duration: duration)
+        lastSample[currentButton] = sample
+        samples.append(sample)
+        guard currentButton != .none, previousButton != currentButton
+        else {
+            return
+        }
+        tapCount += 1
+        previousButton = currentButton
```

Comment:
So... I think that you want to record the previous button whether or not you are incrementing the count, right?
```
         if currentButton != .none, previousButton != currentButton {
             tapCount += 1
         }
         previousButton = currentButton
```